cooperator/models/company.py

```diff
@@ -138,3 +178,78 @@ def onchange_financial_risk_approval_required(self):
     def onchange_generic_rules_approval_required(self):
         if self.generic_rules_approval_required:
             self.display_generic_rules_approval = True
+
+    @api.model
+    def create(self, vals):
+        result = super().create(vals)
+        # The ignore_list is populated such that, if the user defines a template
+        # they want to use during company creation, that choice doesn't get
+        # overridden. The boolean check exists because, when creating a company
+        # from the UI, all empty fields are defined with no value.
+        result._setup_default_cooperator_mail_templates(
+            ignore_list=[key for key, val in vals.items() if val]
+        )
+        return result
+
+    @api.model
+    def _get_cooperator_mail_template_fields(self):
+        return {
+            "cooperator_confirmation_mail_template": "cooperator.email_template_confirmation",
+            "cooperator_confirmation_company_mail_template": (
+                "cooperator.email_template_confirmation_company"
+            ),
+            "cooperator_capital_release_mail_template": (
+                "cooperator.email_template_release_capital"
+            ),
+            "cooperator_waiting_list_mail_template": "cooperator.email_template_waiting_list",
+            "cooperator_certificate_mail_template": "cooperator.email_template_certificat",
+            "cooperator_certificate_increase_mail_template": (
+                "cooperator.email_template_certificat_increase"
+            ),
+            "cooperator_share_transfer_mail_template": (
+                "cooperator.email_template_share_transfer"
+            ),
+            "cooperator_share_update_mail_template": "cooperator.email_template_share_update",
+        }
+
+    def _get_default_cooperator_mail_template(self, xmlid, copy=True):
+        """Get the mail template from its xmlid and either return it or return
+        a copy of it.
+        """
+        self.ensure_one()
+        try:
+            template = self.env.ref(xmlid)
+        except ValueError:
+            return False
+        if copy:
+            result = template.copy()
+            # Set name to 'Company - Template' instead of 'Template (copy)'.
+            result.name = "{} - {}".format(self.name, template.name)
```

Comment:
Shouldn't we also set `company_id` here to the template?
Maybe we need https://github.com/OCA/multi-company/blob/14.0/mail_template_multi_company/models/mail_template.py#L11 for that.
What do you think?

Reply:
Yes, initially I did do that here, but the `mail_template_multi_company` module is not in a usable state as far as I can tell.

https://github.com/OCA/multi-company/issues/483

Reply:
what about removing the default value for `company_id` in this module? would it be too late in that the data would already be assigned to the 1st company?

Reply:
This module = `cooperator` or `mail_template_multi_company`?

Because there's nothing in `cooperator` that we can change to fix the problem described in the linked issue.

Reply:
`mail.template` are not multi-company by default, there isn't a `company_id` in the model if `mail_template_multi_company` is not installed.
If we want to add a unique template for every company, we would need this module, but I understand that the scope of this PR is to add `mail.template` selection for coopertor regardless if they are multi-company or not.